File: 1353-find-resultant-array-after-removing-anagrams/find-resultant-array-after-removing-anagrams.py
class Solution:
    def removeAnagrams(self, words: List[str]) -> List[str]:
        mp = defaultdict(int)
        ans = []
        prev = words[0]

        for word in words:
            srt = "".join(sorted(word))
            if srt not in mp:
                mp[srt] == 1
                ans.append(word)
            elif prev != srt:
                ans.append(word)

            prev = srt

        return ans

File: 1353-find-resultant-array-after-removing-anagrams/find-resultant-array-after-removing-anagrams.cpp
#define ll long long
#define vi vector<int>
#define vll vector<ll>
#define vs vector<string>
#define mpi map<int, int>
#define mps map<string, int>
#define pb push_back
#define all(x) (x).begin(), (x).end()

const int MOD = 1e9 + 7;

class Solution {
public:
    vector<string> removeAnagrams(vector<string>& words) {
        vs ans;
        string prev = "";

        for (auto& word : words) {
            string srt = word;
            sort(all(srt));

            if (srt != prev) {
                ans.pb(word);
            }
            prev = srt;
        }

        return ans;
    }
};